pkg/cmd/roachtest/tests/kv.go

```diff
@@ -170,9 +167,14 @@ func registerKV(r registry.Registry) {
 				sequential = " --sequential"
 			}
 
-			var envFlags string
-			if e := os.Getenv(envKVFlags); e != "" {
-				envFlags = " " + e
+			var duration string
+			envWorkloadDurationFlag := os.Getenv(test.EnvWorkloadDurationFlag)
+			defaultDuration := ifLocal(c, "10s", opts.duration.String())
+
+			if envWorkloadDurationFlag != "" && test.ValidateWorkloadDurationFlag(envWorkloadDurationFlag) {
+				duration += " --duration=" + envWorkloadDurationFlag
+			} else {
+				duration += " --duration=" + defaultDuration
```

Comment:
nit: you can move `defaultDuration := ifLocal(c, "10s", opts.duration.String())` to this scope since it's not used anywhere else.